ansible/roles/ml-data-pipeline/tasks/main.yml

```diff
@@ -0,0 +1,115 @@
+---
+# tasks file for data-pipeline
+- name: Git checkout
+  git:
+    repo: 'https://github.com/Sunbird-Ed/ml-analytics-service.git'
+    dest: "/opt/sparkjobs/ml-analytics-service"
+    version: "{{ ml_analytics_version }}"
+
+- name: CREATE THE USER
+  user:
+    name: "{{ USER }}"
+    state: present
+    home: "/home/{{ USER }}"
+    shell: /bin/bash
+
+- name: INSTALL THE PACKGES
+  apt:  
+    update_cache: yes
+    name: "{{ item }}"
+    state: present
+  loop:
+    - openjdk-8-jdk
+    - software-properties-common
+    - python3-pip
+    - python3.6-dev
+    - python3.6-venv
+    - python-virtualenv
+    - zip
+    - unzip
+
+- name: CHANGE THE OWNERSHIP FOR THIS {{ BASEPATH }} DIRECTORY
+  file:
+    path: "{{ item }}"
+    mode: "0755"
+    recurse: yes
+    state: directory
+    owner: "{{ USER }}"
+    group: "{{ USER }}"
+  loop:
+    - "{{ BASEPATH }}"
+    - "{{ WORKDIR }}"
+    - "{{ WORKDIR }}/faust_as_service"
+
+- name: Change user and create working directory under opt dir and install python virtual environment
+  shell: "cd {{ WORKDIR }} && virtualenv --python=python3.6 spark_venv"
+  become: yes
+  become_user: "{{ USER }}"
+
+- name: Create necessary logs folders for pipeline
+  become: yes
+  file: 
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ USER }}"
+    group: "{{ USER }}"
+    mode: "0755"
+  loop: 
+    - "{{ WORKDIR }}/logs/observation/status"
+    - "{{ WORKDIR }}/logs/observation_evidence"
+    - "{{ WORKDIR }}/logs/project"
+    - "{{ WORKDIR }}/logs/survey"
+    - "{{ WORKDIR }}/logs/survey_evidence"
+
+- name: Initiate virtualenv
```

Comment:
This task name is confusing. Are we install venv or requirements here? Venv is already installed in previous steps rigjt?

Reply:
PIP INSTALLATION OF NECESSARY PACKAGE FROM requirement.txt

Reply:
I am not sure if you understand by what renaming means. Change to something meaningful please!
`name: Install packages from requirement.txt`
Don't resolve comments without making the change!